coder update cli command doesn't explain it needs an arg

# Problem

```
% coder update -h
Update a workspace to the latest template version

Usage:
  coder update [flags]
```

This usage is incorrect, as it needs a workspace name as an arg.

Also, we panic instead of giving an error message when you run it without a workspace name

```
% coder update
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/coder/coder/cli.update.func1(0x1400079e000, {0x10c681598, 0x0, 0x0?})
	/Users/spike/repos/coder/cli/update.go:28 +0x52c
github.com/spf13/cobra.(*Command).execute(0x1400079e000, {0x10c681598, 0x0, 0x0})
	/Users/spike/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:872 +0x4d0
github.com/spf13/cobra.(*Command).ExecuteC(0x14000720c80)
	/Users/spike/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:990 +0x354
main.main()
	/Users/spike/repos/coder/cmd/coder/main.go:18 +0xa8
```

# Acceptance criteria

1. usage should be `coder update <workspace> [flags]`
2. running `coder update` with no args should throw an error (or prompt user for workspace)

--- a/cli/update.go
+++ b/cli/update.go
@@ -18,7 +18,8 @@ func update() *cobra.Command {
 
 	cmd := &cobra.Command{
 		Annotations: workspaceCommand,
-		Use:         "update",
+		Use:         "update <workspace>",
+		Args:        cobra.ExactArgs(1),
 		Short:       "Update a workspace to the latest template version",
 		RunE: func(cmd *cobra.Command, args []string) error {
 			client, err := createClient(cmd)

--- a/cli/update_test.go
+++ b/cli/update_test.go
@@ -1,0 +1,141 @@
+package cli_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/coder/coder/cli/clitest"
+	"github.com/coder/coder/coderd/coderdtest"
+	"github.com/coder/coder/codersdk"
+	"github.com/coder/coder/provisioner/echo"
+	"github.com/coder/coder/pty/ptytest"
+)
+
+func TestUpdate(t *testing.T) {
+	t.Parallel()
+
+	// Test that the function does not panic on missing arg.
+	t.Run("NoArgs", func(t *testing.T) {
+		t.Parallel()
+
+		cmd, _ := clitest.New(t, "update")
+		err := cmd.Execute()
+		require.Error(t, err)
+	})
+
+	t.Run("OK", func(t *testing.T) {
+		t.Parallel()
+
+		client := coderdtest.New(t, &coderdtest.Options{IncludeProvisionerD: true})
+		user := coderdtest.CreateFirstUser(t, client)
+		version1 := coderdtest.CreateTemplateVersion(t, client, user.OrganizationID, nil)
+
+		coderdtest.AwaitTemplateVersionJob(t, client, version1.ID)
+		template := coderdtest.CreateTemplate(t, client, user.OrganizationID, version1.ID)
+
+		cmd, root := clitest.New(t, "create",
+			"my-workspace",
+			"--template", template.Name,
+			"-y",
+		)
+		clitest.SetupConfig(t, client, root)
+
+		err := cmd.Execute()
+		require.NoError(t, err)
+
+		ws, err := client.WorkspaceByOwnerAndName(context.Background(), "testuser", "my-workspace", codersdk.WorkspaceOptions{})
+		require.NoError(t, err)
+		require.Equal(t, version1.ID.String(), ws.LatestBuild.TemplateVersionID.String())
+
+		version2 := coderdtest.UpdateTemplateVersion(t, client, user.OrganizationID, &echo.Responses{
+			Parse:           echo.ParseComplete,
+			Provision:       echo.ProvisionComplete,
+			ProvisionDryRun: echo.ProvisionComplete,
+		}, template.ID)
+		_ = coderdtest.AwaitTemplateVersionJob(t, client, version2.ID)
+
+		err = client.UpdateActiveTemplateVersion(context.Background(), template.ID, codersdk.UpdateActiveTemplateVersion{
+			ID: version2.ID,
+		})
+		require.NoError(t, err)
+
+		cmd, root = clitest.New(t, "update", ws.Name)
+		clitest.SetupConfig(t, client, root)
+
+		err = cmd.Execute()
+		require.NoError(t, err)
+
+		ws, err = client.WorkspaceByOwnerAndName(context.Background(), "testuser", "my-workspace", codersdk.WorkspaceOptions{})
+		require.NoError(t, err)
+		require.Equal(t, version2.ID.String(), ws.LatestBuild.TemplateVersionID.String())
+	})
+
+	t.Run("WithParameter", func(t *testing.T) {
+		t.Parallel()
+
+		client := coderdtest.New(t, &coderdtest.Options{IncludeProvisionerD: true})
+		user := coderdtest.CreateFirstUser(t, client)
+		version1 := coderdtest.CreateTemplateVersion(t, client, user.OrganizationID, nil)
+
+		coderdtest.AwaitTemplateVersionJob(t, client, version1.ID)
+		template := coderdtest.CreateTemplate(t, client, user.OrganizationID, version1.ID)
+
+		cmd, root := clitest.New(t, "create",
+			"my-workspace",
+			"--template", template.Name,
+			"-y",
+		)
+		clitest.SetupConfig(t, client, root)
+
+		err := cmd.Execute()
+		require.NoError(t, err)
+
+		ws, err := client.WorkspaceByOwnerAndName(context.Background(), "testuser", "my-workspace", codersdk.WorkspaceOptions{})
+		require.NoError(t, err)
+		require.Equal(t, version1.ID.String(), ws.LatestBuild.TemplateVersionID.String())
+
+		defaultValue := "something"
+		version2 := coderdtest.UpdateTemplateVersion(t, client, user.OrganizationID, &echo.Responses{
+			Parse:           createTestParseResponseWithDefault(defaultValue),
+			Provision:       echo.ProvisionComplete,
+			ProvisionDryRun: echo.ProvisionComplete,
+		}, template.ID)
+		coderdtest.AwaitTemplateVersionJob(t, client, version2.ID)
+
+		err = client.UpdateActiveTemplateVersion(context.Background(), template.ID, codersdk.UpdateActiveTemplateVersion{
+			ID: version2.ID,
+		})
+		require.NoError(t, err)
+
+		cmd, root = clitest.New(t, "update", ws.Name)
+		clitest.SetupConfig(t, client, root)
+
+		pty := ptytest.New(t)
+		cmd.SetIn(pty.Input())
+		cmd.SetOut(pty.Output())
+
+		doneChan := make(chan struct{})
+		go func() {
+			defer close(doneChan)
+			err := cmd.Execute()
+			assert.NoError(t, err)
+		}()
+
+		matches := []string{
+			fmt.Sprintf("Enter a value (default: %q):", defaultValue), "bingo",
+			"Enter a value:", "boingo",
+		}
+		for i := 0; i < len(matches); i += 2 {
+			match := matches[i]
+			value := matches[i+1]
+			pty.ExpectMatch(match)
+			pty.WriteLine(value)
+		}
+
+		<-doneChan
+	})
+}

